docker/compute_worker/compute_worker.py

```diff
@@ -607,7 +602,41 @@ def start(self):
             self.watch_file(),
             loop=loop,
         )
-        loop.run_until_complete(gathered_tasks)
+        signal.signal(signal.SIGALRM, alarm_handler)
+        signal.alarm(self.execution_time_limit)
+        try:
+            loop.run_until_complete(gathered_tasks)
+        except ExecutionTimeLimitExceeded:
+            pass
```

Comment:
instead of passing here we should let the user know why it failed i suppose?

```python
self._update_status(STATUS_FAILED, f"Time limit exceeded. Limit was {self.execution_time_limit}ms.")
```

this might get overwritten somewhere else in the script though?

Reply:
`raise SubmissionException(f"Execution Time Limit exceeded. Limit was {self.execution_time_limit} seconds")` as a solution instead, since we catch a Submission Exception to call `_update_status(STATUS_FAILED)`